security/src/main/java/pm/gnosis/svalinn/security/impls/AndroidFingerprintHelper.kt

```diff
@@ -68,53 +71,35 @@ class AndroidFingerprintHelper(private val context: Context) : FingerprintHelper
                 else init(Cipher.DECRYPT_MODE, getOrCreateKey(), IvParameterSpec(iv))
             }
 
-    override fun authenticate(iv: ByteArray?): Observable<AuthenticationResult> =
-        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.M) Observable.error(FingerprintNotAvailable("Android version not supported (${Build.VERSION.SDK_INT})"))
-        else Observable.create(AuthenticateObservable(context, iv, ::createCipher))
-
-    private class AuthenticateObservable(
-        context: Context, private val iv: ByteArray? = null,
-        private val cipherProvider: (ByteArray?) -> Cipher
-    ) : FingerprintManagerCompat.AuthenticationCallback(), ObservableOnSubscribe<AuthenticationResult> {
-
-        private val fingerprintManager = FingerprintManagerCompat.from(context)
-        private var emitter: ObservableEmitter<AuthenticationResult>? = null
-
-        override fun subscribe(emitter: ObservableEmitter<AuthenticationResult>) {
-            try {
-                val cryptoObject = FingerprintManagerCompat.CryptoObject(cipherProvider(iv))
-                this.emitter = emitter
-                val signal = CancellationSignal()
-                fingerprintManager.authenticate(cryptoObject, 0, signal, this, null)
-                emitter.setCancellable {
-                    if (!signal.isCanceled) {
-                        signal.cancel()
-                        this.emitter = null
+    @SuppressLint("MissingPermission")
+    @RequiresApi(Build.VERSION_CODES.M)
+    override suspend fun authenticate(iv: ByteArray?): AuthenticationResult {
+        val cryptoObject = FingerprintManagerCompat.CryptoObject(createCipher(iv))
+        val fingerprintManager = FingerprintManagerCompat.from(context)
+        val signal = CancellationSignal()
```

Comment:
I would inline this as it is only used in `onAuthenticationError` to keep the initialisation of stuff as minimal as necessary.

Reply:
Yeah cancelation is not handled right now ... this is a todo

Reply:
Is `authenticate()`one of the methods to disable? We're not using the FingerPrinHelper ATM. If not. I dont't understand what exactly is wrong with the cancellation.